# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Srichandana2404/flyrank-ml-track/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
## My Lane


**Lane:** Refresh / Content Opportunity Scoring  
**Why:** The dataset shows which pages are declining or stale.
 Ranking them helps editors decide which ones to refresh first, so limited effort goes to the most valuable pages.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## The Question

**Decision:** Which pages should be reviewed first for refresh.  
**Action:** Editors update, expand, or prune those pages.  
**Cost of a wrong call:** Wasting effort on low‑impact pages, missing high‑value opportunities, or even harming traffic if a strong page is pruned.  
**Who acts on it:** Content editors and SEO teams.  
**Why ML helps:** The dataset has impressions, clicks, sessions, and CTR. Machine learning can rank pages more effectively than fixed rules, helping editors focus on the most promising candidates.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

# Load the starter dataset
df = pd.read_csv("content_refresh_anonymized.csv")

# Show basic numbers
print("Rows in dataset:", len(df))
print("Average impressions (90d):", df["impressions_90d"].mean())
print("Pages with declining trend:", (df["trend_direction"]=="down").sum())


Rows in dataset: 30000
Average impressions (90d): 5200.3663
Pages with declining trend: 16262


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Quick Look at the Data

- The dataset has about 30K rows.  
- The average number of impressions in 90 days is around 5200.  
- About 16262 pages show a declining trend.  

These numbers show that the dataset is large enough to be meaningful, and that many pages are clearly declining. This makes the "Refresh / Content Opportunity Scoring" lane worth pursuing, because it helps prioritize which pages should be reviewed first.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Careful Words

What I can claim:
- I can observe patterns in the data (impressions, clicks, sessions, CTR).  
- I can provide directional insights and ranked recommendations to guide editors.  
- My work is decision-support: it helps prioritize which pages to review first.

What I cannot claim:
- I cannot prove causation (for example, that refreshing a page will guarantee recovery).  
- I cannot predict or explain Google’s algorithm directly.  
- I cannot make perfect forecasts — results are based on observed signals, not certainty.



## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.